new reveal panel is added

Yasintrm opened the pull request:
This PR adds reveal panel component to the tyk-ui library (Tests will be added later).

Please check the screencast for how it behaves.

![tyk ui reveal](https://github.com/TykTechnologies/tyk-ui/assets/1834735/9c2e765e-28c5-441f-b007-1e2977987ce1)

ifrim commented on lib/sass/components/RevealPanel/sass/RevealPanel.scss:
```diff
@@ -0,0 +1,85 @@
+    &__gutter {
+        height: 2rem;
+        width: 100%;
+        cursor: row-resize !important;
```
do we need this `!important` here?

ifrim reviewed (commented):
In my opinion this looks a bit over engineered.
It has a controller which needs to be instantiated.
It has its own pub/sub system. Which is not even used since you only give it one callback to call.
It has bitwise operations.
The overall style seems more closely to jquery than react.

In the end it should be a simple component where you change the height of a container, and that has some css to make the container stick to one side of its parent. It doesn't have too much complexity and it shouldn't have.
I saw some other code on previous reviews, what happened to that code :)

Yasintrm commented:
> In my opinion this looks a bit over engineered. It has a controller which needs to be instantiated. It has its own pub/sub system. Which is not even used since you only give it one callback to call. It has bitwise operations. The overall style seems more closely to jquery than react.
> 
> In the end it should be a simple component where you change the height of a container, and that has some css to make the container stick to one side of its parent. It doesn't have too much complexity and it shouldn't have. I saw some other code on previous reviews, what happened to that code :)

:D  yes it started simple. Then I needed another component to listen height changes too. So I added that event listeners because of that. Bitwise code block was a copy paste from stackoverflow. It had a bug when mouse down is clicked and you start moving sometimes it keeps dragging the panel even if the mouse button is not down. So I added primary mouse button check there (code block that has bitwise op). Every code there is added because I encountered a bug or necessity so it is why a little bit different than the one you see :).

Yasintrm commented:
I just pushed a new commit: bitwise operation  and controller instantiation are removed.
